cosmos/plugin/af3_plugin_impl.py

```diff
@@ -0,0 +1,293 @@
+    env_name = f"AIRFLOW_CONN_{conn_id.upper()}"  # type: ignore[union-attr]
+    env_value = conn.get_uri()
+    with patch.dict(os.environ, {env_name: env_value}, clear=False):
+        yield
+
+
+def _read_content_via_object_storage(path: str, conn_id: str | None = None) -> Any:
+    with connection_env(conn_id):
+        p = ObjectStoragePath(path, conn_id=conn_id) if conn_id else ObjectStoragePath(path)
+        with p.open("r") as f:  # type: ignore[no-untyped-call]
+            content = f.read()  # type: ignore[no-any-return]
+        return content
+
+
+def open_file(path: str, conn_id: str | None = None) -> Any:
+    """
+    Retrieve a file from http, https, gs, s3, or wasb.
+
+    Raise a (base Python) FileNotFoundError if the file is not found.
+    """
+    if path.strip().startswith(("s3://", "gs://", "gcs://", "wasb://", "abfs://", "az://", "http://", "https://")):
+        return _read_content_via_object_storage(path, conn_id=conn_id)
+    else:
+        with open(path) as f:
+            content = f.read()
+        return content  # type: ignore[no-any-return]
+
+
+iframe_script = """
+<script>
+  // Prevent parent hash changes from sending a message back to the parent.
+  // This is necessary for making sure the browser back button works properly.
+  let hashChangeLock = true;
+
+  window.addEventListener('hashchange', function () {
+    if (!hashChangeLock) {
+      window.parent.postMessage(window.location.hash);
+    }
+    hashChangeLock = false;
+  });
+
+  window.addEventListener('message', function (event) {
+    let msgData = event.data;
+    if (typeof msgData === 'string' && msgData.startsWith('#!')) {
+      let updateUrl = new URL(window.location);
+      updateUrl.hash = msgData;
+      hashChangeLock = true;
+      history.replaceState(null, null, updateUrl);
+    }
+  });
+</script>
+"""
+
+
+def _load_projects_from_conf() -> dict[str, dict[str, Optional[str]]]:
+    """
+    Load dbt docs projects configuration.
+
+    Supports either:
+    - [cosmos] dbt_docs_projects = JSON mapping of project slug to {"dir","conn_id","index"}
+    - Legacy single-project settings: dbt_docs_dir, dbt_docs_conn_id, dbt_docs_index_file_name
+    """
+    projects_raw = conf.get("cosmos", "dbt_docs_projects", fallback=None)
+    projects: dict[str, dict[str, Optional[str]]] = {}
+    if projects_raw:
+        try:
+            parsed = json.loads(projects_raw)
+            if isinstance(parsed, dict):
+                for key, value in parsed.items():
+                    if not isinstance(value, dict):  # pragma: no cover
+                        continue
+                    projects[str(key)] = {
+                        "dir": value.get("dir"),
+                        "conn_id": value.get("conn_id"),
+                        "index": value.get("index", "index.html"),
+                        "name": value.get("name"),
+                    }
+        except Exception:
+            # Ignore malformed config; fall back to legacy
```

Comment:
If customers attempt to use the new `dbt_docs_projects` and its configuration has issues, I believe we should not fallback to the old configuration. We should be explicit with users about the error and provide clear instructions so they can fix it. If they want to fallback to the old way, they can do this explicitly.